core/scripts/umip-3/VoteSimulator.js

```diff
@@ -0,0 +1,173 @@
+  let snapshot = await takeSnapshot(web3);
+  snapshotId = snapshot["result"];
+  console.log("Snapshotting starting state...", snapshotId);
+
+  /** *********************************
+   * 0) Initial setup                *
+   ***********************************/
+
+  console.log("0. SETUP PHASE");
+  const voting = await Voting.deployed();
+  const governor = await Governor.deployed();
+
+  let currentTime = (await voting.getCurrentTime()).toNumber();
+  let votingPhase = (await voting.getVotePhase()).toNumber();
+
+  const secondsPerDay = 86400;
+  const accounts = await web3.eth.getAccounts();
+
+  // These two assertions must be true for the script to be starting in the right state. This assumes the script
+  // is run within the same round and phase as running the `Propose.js` script.
+  assert.equal((await governor.numProposals()).toNumber(), 1); // there should be 1 proposal
+  assert.equal((await voting.getPendingRequests()).length, 0); // There should be no pending requests
+
+  console.log(
+    "1 pending proposal. No pending requests.\nCurrent timestamp:",
+    currentTime,
+    "Voting phase:",
+    votingPhase,
+    "CurrentRoundId",
+    (await voting.getCurrentRoundId()).toString()
+  );
+
+  /** ****************************************************
+   * 1) Advance Ganache block time to move voting round *
+   ******************************************************/
+
+  console.log("1. TIME ADVANCE FOR VOTING");
+
+  // If the DVM is currently in the commit phase (0) then we need to advance into the next round by advancing
+  // forward by 2 days. Else if we are in the reveal phase (1) then we need to advance into the next round by
+  // advancing by one day to take us into the next round.
+  if (votingPhase == 0) {
+    await advanceBlockAndSetTime(web3, currentTime + secondsPerDay * 2);
+  } else {
+    await advanceBlockAndSetTime(web3, currentTime + secondsPerDay);
+  }
+  console.log(
+    "⏱  Advancing time by one phase to enable voting by the DVM\nNew timestamp",
+    (await voting.getCurrentTime()).toString(),
+    "voting phase",
+    (await voting.getVotePhase()).toNumber(),
+    "CurrentRoundId",
+    (await voting.getCurrentRoundId()).toString()
+  );
+  let pendingRequests = await voting.getPendingRequests();
+  assert.equal(pendingRequests.length, 1); // the one proposal should have advanced to a request
+
+  /** *****************************************************
+   * 2) Build vote tx from the foundation wallet         *
+   *******************************************************/
+
+  let currentRoundId = (await voting.getCurrentRoundId()).toString();
+
+  const identifier = pendingRequests[0].identifier.toString();
+  const time = pendingRequests[0].time.toString();
+  const price = "1"; // a "yes" vote
+
+  const salt = getRandomUnsignedInt();
+  // Main net DVM uses the old commit reveal scheme of hashed concatenation of price and salt
+  const voteHash = web3.utils.soliditySha3(price, salt);
```

Comment:
So to be clear, this is a useless vote right? The proposal doesn't do anything in other words.

Would it give us more confidence to submit a proposal that has some transactions that actually change the DVM state such as setting fees for example, just so we can test the proposals change DVM state like they should?

Reply:
Ah, I see. This script just simulates the vote on the proposal that is crafted in a previous script.